pandas/tests/groupby/test_groupby_shift_diff.py

```diff
@@ -55,7 +55,7 @@ def test_group_shift_with_fill_value():
         columns=["Z"],
         index=None,
     )
-    result = g.shift(-1, fill_value=0)[["Z"]]
+    result = g.shift(-1, fill_value=0)
```

Comment:
do we have sufficient coverage of the tests from the OP?

Reply:
Yes. I have investigated it. 

The original problem was that the grouping columns were improperly returned. The old form of this test would extract the value column (`[["Z"]]`), so the test did not detect that the grouping columns were also returned. I have modified this test so it no longer extracts the value column; if the grouping columns are returned (the OP) this test will now fail.

Reply:
kk great

if u can resolve conflicts and merge master

Reply:
@jreback conflicts resolved. I think it can be merged.